sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -10,53 +10,45 @@ import { AbortSignalLike } from '@azure/abort-controller';
 export type CancelOnProgress = () => void;
 
 // @public
-export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): Promise<SimplePollerLike<TState, TResult>>;
+export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): PollerLike<TState, TResult>;
 
 // @public
 export interface CreateHttpPollerOptions<TResult, TState> {
     intervalInMs?: number;
-    processResult?: (result: unknown, state: TState) => TResult;
+    processResult?: (result: unknown, state: TState) => TResult | Promise<TResult>;
```

Comment:
Why is the return typue `TResult | Promise<TResult>` and not just `Promise<TResult>`?

Reply:
The change is from Modular we would leverage this function to pass our deserilization function which is designed by async one. 
https://github.com/Azure/autorest.typescript/pull/2198/files#diff-24bf7d5f60df1236ef347c40211b8f14bb86ece0f206a7161417019e3486e955R189.

But if we would also accept other downstreamings It's better to have the capability to accept both sync and async way to process the final result and not introducing breakings(I would assume in most of times just synced processResult could handle any transformation).